src/components/SocialBar/SocialBarIcon.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+
+import styles from './SocialBarIcon.st.css';
+import { Tooltip } from '../Tooltip';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { ButtonProps } from 'wix-ui-core/dist/src/components/button-next/button-next';
+import { TPAComponentProps } from '../../types';
+import { IconButton, Skins } from '../IconButton';
+import { SocialBarInjectedProps } from './SocialBar';
+
+interface SocialBarIconProps
+  extends ButtonProps,
+    TPAComponentProps,
+    SocialBarInjectedProps {
+  tooltip?: string;
+  icon: React.ReactNode;
+}
+
+export class SocialBarIcon extends React.Component<SocialBarIconProps> {
+  static defaultProps = { socialBarTheme: 'light' };
+
+  render() {
+    const { children, tooltip, socialBarTheme, ...rest } = this.props;
+    return (
+      <TPAComponentsConsumer>
+        {({ mobile }) => (
+          <div {...styles('root', {}, rest)}>
+            <Tooltip
+              appendTo={'scrollParent'}
+              content={tooltip}
+              placement="bottom"
+              disabled={!tooltip || mobile}
+            >
+              <IconButton
+                {...styles('iconButton', { theme: socialBarTheme })}
```

Comment:
You should pass a className instead of destructuring the styles object.
i.e.:
```
<IconButton className={styles.iconButton} />
```
and pass the state to the root

Reply:
moved all child states to root